src/features/comments.ts

```diff
@@ -0,0 +1,82 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment } from '../types/Comment';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+
+type CommentsState = {
+  items: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: CommentsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadComments = createAsyncThunk(
+  'comments/fetch',
+  async (postId: number) => {
+    const result = await getPostComments(postId);
+
+    return result;
+  },
+);
+
+export const addComment = createAsyncThunk(
+  'comments/post',
+  async (data: Omit<Comment, 'id'>) => {
+    const result = await createComment(data);
+
+    return result;
+  },
+);
+
+export const removeComment = createAsyncThunk(
+  'comments/delete',
+  async (commentId: number) => {
+    await deleteComment(commentId);
+
+    return commentId;
+  },
+);
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(loadComments.pending, state => {
+      return { ...state, loaded: false };
+    });
+
+    builder.addCase(loadComments.fulfilled, (state, actions) => {
+      return { ...state, items: actions.payload, loaded: true };
+    });
+
+    builder.addCase(loadComments.rejected, state => {
+      return { ...state, hasError: true, loaded: true };
+    });
```

Comment:
In the `loadComments.rejected` case, you're setting `hasError` to `true`, which is good, but you might also want to reset `items` to an empty array to ensure that any previously loaded comments are not displayed when an error occurs.